Implements martials_like for cross-mod martial arts compatibility (#48905)
CleverRaven/Cataclysm-DDA: doc/JSON_INFO.md

"price": 100,                                // Used when bartering with NPCs. For stackable items (ammo, comestibles) this is the price for stack_size charges. Can use string "cent" "USD" or "kUSD".
"price_postapoc": "1 USD",                       // Same as price but represent value post cataclysm. Can use string "cent" "USD" or "kUSD".
"material": ["COTTON"],                      // Material types, can be as many as you want.  See materials.json for possible options

> "martials_like": ["weaponid1"]               // (Optional) This weapon can be used in martial arts that use the weapons in the list.

Reviewer: ```suggestion
"martials_like": ["weaponid1"]               // (Optional) This weapon can be used in martial arts that use the weapons in the list. Not recursive, the weapons specified must be explicitly listed in martial art.
```

> "price_postapoc": "1 USD",                       // Same as price but represent value post cataclysm. Can use string "cent" "USD" or "kUSD".
> "material": ["COTTON"],                      // Material types, can be as many as you want.  See materials.json for possible options
> "martials_like": ["weaponid1"]               // (Optional) This weapon can be used in martial arts that use the weapons in the list. Not recursive, the weapons specified must be explicitly listed in martial art.

Reviewer 2: ```suggestion
"price_postapoc": "1 USD",                       // Same as price but represents value post cataclysm. Can use string "cent" "USD" or "kUSD".
"material": ["COTTON"],                      // Material types, can be as many as you want.  See materials.json for possible options.
"martials_like": ["weaponid1"]               // (Optional) This weapon can be used in martial arts that use the (other) weapons in the list. Not recursive; the weapons specified must be explicitly listed by the martial art.
```

"material": ["iron", "wood"], // Material types.  See materials.json for possible options
"weight": 907,         // Weight, measured in grams
"volume": "1500 ml",   // Volume, volume in ml and L can be used - "50 ml" or "2 L"

> "martials_like": ["weaponid1"] // This weapon can be used in martial arts that use the weapons in the list.

Reviewer: ```suggestion
"martials_like": ["weaponid1"] // This weapon can be used in martial arts that use the weapons in the list.  Not recursive, the weapons specified must be explicitly listed in martial art.
```

> "martials_like": ["weaponid1"] // This weapon can be used in martial arts that use the weapons in the list.  Not recursive, the weapons specified must be explicitly listed in martial art.

Reviewer 2: ```suggestion
"martials_like": ["weaponid1"] // This weapon can be used in martial arts that use the (other) weapons in the list.  Not recursive; the weapons specified must be explicitly listed by the martial art.
```
